example/generator/odp_generator.c

```diff
@@ -1210,10 +1214,35 @@ int main(int argc, char *argv[])
 	/* Configure scheduler */
 	odp_schedule_config(NULL);
 
+	if (odp_pool_capability(&pool_capa)) {
+		ODPH_ERR("Error: pool capability failed\n");
+		return -1;
+	}
+
+	pkt_len = (args->appl.payload + MAX_HDR_LEN) > POOL_PKT_LEN ?
+			(args->appl.payload + MAX_HDR_LEN) : POOL_PKT_LEN;
+
+	if (pool_capa.pkt.max_len && pkt_len > pool_capa.pkt.max_len)
+		pkt_len = pool_capa.pkt.max_len;
+
+	/* Check weather we have sufficient segments to support requested packet
+	 * length, if not adjust to bigger segment size */
+	if ((seg_len * pool_capa.pkt.max_segs_per_pkt) < pkt_len)
+		seg_len = pkt_len / pool_capa.pkt.max_segs_per_pkt;
+
+	if (pool_capa.pkt.min_seg_len && seg_len < pool_capa.pkt.min_seg_len)
+		seg_len = pool_capa.pkt.min_seg_len;
+
+	if (pool_capa.pkt.max_seg_len && seg_len > pool_capa.pkt.max_seg_len)
+		seg_len = pool_capa.pkt.max_seg_len;
```

Comment:
This is wrong. This application makes the assumption that certain amount of packet data is contiguous in memory. If the assumption is not valid, the application overwrites memory which it does not own and which may contain e.g. internal packet metadata of the underlying ODP implementation. If the underlying ODP does not support long enough contiguous data (big enough max_seg_len), one cannot just ignore the assumption. In that case the application should just exit as it cannot create the kind of pool that it needs.

Reply:
Will remove this code